Fabric.Terminology.Domain/DomainExtensions.String.cs

```diff
@@ -1,13 +1,15 @@
 ﻿namespace Fabric.Terminology.Domain
 {
+    using NullGuard;
+
     /// <summary>
     /// Extension methods for <see>
     ///         <cref>System.String</cref>
     ///     </see>
     /// </summary>
-    public static partial class Extensions
+    public static partial class DomainExtensions
     {
-        public static bool IsNullOrWhiteSpace(this string value)
+        public static bool IsNullOrWhiteSpace([AllowNull] this string value)
         {
             return (value == null) || (value.Trim().Length == 0);
```

Comment:
How is this different from `string.IsNullOrWhitespace()`?

Reply:
Do you mean `string.IsNullOrEmpty()`? If so, this is intended to catch non whitespace characters - but to be honest, I've just ported this method around for years and find myself typing it rather than .IsNullOrEmpty ... can be removed.

Reply:
No, I meant [`string.IsNullOrWhitespace(...)`](https://msdn.microsoft.com/en-us/library/system.string.isnullorwhitespace(v=vs.110).aspx). That method was added in .NET 4. At the very least, it seems like this extension method could pass through to that method. It might be worth considering just inlining it at that point.